Code/GraphMol/RascalMCES/RascalMCES.cpp

```diff
@@ -350,8 +350,27 @@ bool checkAromaticRings(const ROMol &mol1,
   if (!mol1Bond->getIsAromatic() || !mol2Bond->getIsAromatic()) {
     return true;
   }
+
+  auto isInRing = [](const std::vector<std::vector<int>> &rings,
+                     int bondIdx) -> bool {
+    for (const auto &r : rings) {
+      if (std::find(r.begin(), r.end(), bondIdx) != r.end()) {
+        return true;
+      }
+    }
+    return false;
+  };
+
+  // If neither bond was in a ring, but they were marked aromatic, then
+  // the two mols are fragments so it's ok to match these bonds.
   const auto &mol1BondRings = mol1.getRingInfo()->bondRings();
   const auto &mol2BondRings = mol2.getRingInfo()->bondRings();
+  auto mol1BondInRing = isInRing(mol1BondRings, mol1BondIdx);
+  auto mol2BondInRing = isInRing(mol2BondRings, mol2BondIdx);
```

Comment:
I think this would be faster as:
```suggestion
  bool mol1BondInRing = mol1.getRingInfo()->numBondRings(mol1BondIdx);
  bool mol2BondInRing = mol2.getRingInfo()->numBondRings(mol2BondIdx);
```

Reply:
It's probably going to be impossible to tell if it's faster or not, as it won't be used very often.  Using existing code looks more competent :-).